keps/140-scenario-based-simulation/README.md

```diff
@@ -43,368 +64,539 @@ Therefore, this kep proposes to define the scenario **as CRD**. All clients, inc
+At (1), .spec.operations is performed and preparing controllers may perform some operations to move the current cluster state closer to the desired state.
 
-> etcd is designed to handle small key value pairs typical for metadata. Larger requests will work, but may increase the latency of other requests. By default, the maximum size of any request is 1.5 MiB. This limit is configurable through --max-request-bytes flag for etcd server.
-https://etcd.io/docs/v3.4/dev-guide/limit/#request-size-limit
+And **the simulated controllers need to stop working while the preparing controllers work.** (Actually, the simulated controller has been stopped since (1))
+In .status.Result.Timeline, Preparing controllers actions are treated as if they were instantaneous.
 
-And, sometimes, users may want to reduce this updating request more for some reasons. 
+So, why..? This is because we need to ensure that the simulation results do not vary significantly from run to run. 
 
-For example, 
-- when watching Scenario for dynamic event addition 
-- when using Scenario for accurate benchmark testing, users may want to reduce the request to update Scenario for kube-apiserver as much as possible
+For example:
+The user has the CRD named "NodeSet". It's like ReplicaSet, but literally, it creates a defined number of Nodes.
+Let's say the user wants to simulate the scheduler and defines the operation in the Scenario that creates NodeSet to create 1000 Nodes. Since it is strictly impossible to create 1000 Nodes simultaneously, the scheduler will schedule pending Pods to the Nodes created first. 
+And depending on how fast the NodeSet controller creates the 1000 Node, the simulation results will change.
+The scheduler needs to be stopped scheduling until the NodeSet controller creates 1000 Nodes so that the speed of the NodeSet controller doesn't affect the simulation result.
 
-We can add a new configuration environment variable `UPDATE_SCENARIO_RESULTS_STRATEGY` and define some strategy like:
-- `UPDATE_SCENARIO_RESULTS_STRATEGY=AtMovingNextStep`: default value. update status.scenarioResult in Scenario resource when proceeding to the next ScenarioStep.
-- `UPDATE_SCENARIO_RESULTS_STRATEGY=OnPause`: update status.scenarioResult in Scenario resource when the Scenario's phase becomes `Paused`, `Succeeded` or `Failed`.
-- `UPDATE_SCENARIO_RESULTS_STRATEGY=OnDone`: update status.scenarioResult in Scenario resource when the Scenario's phase becomes `Succeeded` or `Failed`.
+##### 3. preparing controllers finish to work 
+
+ScenarioStep: {X, 0}
+StepPhase: OperatingCompleted
+
+See [#Expect when controllers finish working](#Expect-when-controllers-finish-working).
+
+We use `WaitForControllerFinish` to wait for all controllers to finish their work to deal with (1).
+
+##### 4. the simulated controller starts
+
+ScenarioStep: {X, 0}
+StepPhase: ControllerRunning
+
+The simulated controllers have been stopped until now.
+
+Here, start one of the simulated controllers.
```

Comment:
In which order if the Scenario has multiple simulated controllers?? We should describe it here.

Reply:
> These are run one by one in the same order specified in Enabled field.

It's described in L95, but yeah, I'd add the note for it here as well.